semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -640,7 +726,9 @@ and check_tainted_expr env exp : Taints.t * Lval_env.t =
       (Taints.empty, lval_env)
   | None ->
       let sinks =
-        orig_is_sink env.config exp.eorig |> Common.map trace_of_match
+        orig_is_sink env.config exp.eorig
```

Comment:
Are there other uses of `orig_is_sink`? It seems like this is at least the most common use, could we just make it so that `orig_is_sink` defaults to filtering for `best_match` and make the special case not filtering?